datacube/utils/geometry/_base.py

```diff
@@ -116,25 +116,34 @@ def from_points(p1: Tuple[float, float],
                                    (p1[1], p2[1]))
 
 
-def _make_crs_key(crs_spec: Union[str, _CRS]) -> str:
+def _make_crs_key(crs_spec: Union[str, int, _CRS]) -> str:
     if isinstance(crs_spec, str):
         normed_epsg = crs_spec.upper()
         if normed_epsg.startswith("EPSG:"):
             return normed_epsg
         return crs_spec
+    if isinstance(crs_spec, int):
+        return f"EPSG:{crs_spec}"
     return crs_spec.to_wkt()
 
 
 @cachetools.cached({}, key=_make_crs_key)  # type: ignore[misc]
-def _make_crs(crs: Union[str, _CRS]) -> Tuple[_CRS, str, Optional[int]]:
-    if isinstance(crs, str):
-        crs = _CRS.from_user_input(crs)
-    epsg = crs.to_epsg()
-    if epsg is not None:
-        crs_str = f"EPSG:{epsg}"
-    else:
+def _make_crs(crs: Union[str, int, _CRS]) -> Tuple[_CRS, str, Optional[int]]:
+    epsg = False
+    crs_str = None
+    if isinstance(crs, int):
+        crs_str = f"EPSG:{crs}"
+        epsg = crs
+    elif isinstance(crs, str):
+        if crs.upper().startswith("EPSG:"):
+            crs_str = crs.upper()
+            epsg = int(crs_str.split(":", maxsplit=1)[-1])
+        else:
+            crs_str = crs
+    crs = _CRS.from_user_input(crs)
```

Comment:
Always call `_CRS.from_user_input` as it supports all of these types. Also, if it is a `_CRS`, it just returns the original `_CRS` [ref](https://github.com/pyproj4/pyproj/blob/837d965655d87351e493491885f289370a1e31dd/pyproj/crs/crs.py#L499-L500).